pkg/sql/catalog/lease/lease_test.go

```diff
@@ -405,26 +406,20 @@ func TestLeaseManagerReacquire(testingT *testing.T) {
 	t.expectLeases(descID, "/1/1")
 	e1 := l1.Expiration()
 
-	// Another lease acquisition from the same node will result in a new lease.
-	rt := removalTracker.TrackRemoval(l1.Underlying())
+	// Another lease acquisition from the same node will result in getting the
+	// same lease again, since it's tied to the session ID
 	l3 := t.mustAcquire(1, descID)
 	e3 := l3.Expiration()
-	if l1.Underlying().GetID() == l3.Underlying().GetID() && e3.WallTime == e1.WallTime {
-		t.Fatalf("expected different leases, but found %v", l1)
-	}
-	if e3.WallTime < e1.WallTime {
-		t.Fatalf("expected new lease expiration (%s) to be after old lease expiration (%s)",
-			e3, e1)
-	}
-	// In acquiring the new lease the older lease is released.
-	if err := rt.WaitForRemoval(); err != nil {
-		t.Fatal(err)
+	if l1.Underlying().GetID() == l3.Underlying().GetID() && e3.WallTime != e1.WallTime {
+		t.Fatalf("expected same leases because the session time will be the same, but found %v", l1)
```

Comment:
I didn't quite understand this test, both old version or this new version.

Old version: when we re-acquire a lease on the same table, why do we assert we get a different lease (one with a later expiration time)?

New version: when we re-acquire a lease on the same table, why do we assert we get a lease with the same expiration? Couldn't the sqlliveness table for this particular seesion/node get heartbeaten and update the expiration of this session/node, and thus the re-acquired lease will have a later expiration? Or, this is impossible bc we always read the sqlliveness table at one particular timestamp both the first and the second time we acquire a lease, and thus we should expect to see the same expiration of the same session, and therefore same lease expiration?

Do you think you can help me understand this test, both old version and new version?

Reply:
note: during offline discussion, we need to "stabilize" the test by ensuring between first acquire and second acquire, the session expiration (on sqlliveness table) does not change (i.e. it's not heartbeaten). We can do this by, maybe, using a fake sqlliveness provider to ensure the session expiration data is unchanged.